rasa/multi_skill.py

```diff
@@ -0,0 +1,148 @@
+import logging
+from typing import Text, List, Union, Set, Dict, Optional
+import os
+
+import rasa.utils.io as io_utils
+from rasa import data
+
+logger = logging.getLogger(__name__)
+
+
+class SkillSelector:
+    def __init__(self, imports: Set[Text], project_directory: Optional[Text] = None):
+        self._imports = imports
+        self._project_directory = project_directory
+
+    @classmethod
+    def empty(cls) -> "SkillSelector":
+        return cls(set())
+
+    @classmethod
+    def load(cls, config: Text) -> "SkillSelector":
+        """
+        Loads the specification from the config files.
+        Args:
+            config: Path to the root configuration file in the project directory.
+
+        Returns:
+            `SkillSelector` which specifies the loaded skills.
+        """
+        # All imports are by default relative to the root config file directory
+        config = os.path.abspath(config)
+        selector = cls(set(), os.path.dirname(config))
+        selector = cls._from_file(config, selector)
+
+        logger.debug("Selected skills: {}.".format(selector._imports))
+
+        return selector
+
+    @classmethod
+    def _from_path(cls, path: Text, skill_selector: "SkillSelector") -> "SkillSelector":
+        if os.path.isfile(path):
+            return cls._from_file(path, skill_selector)
+        elif os.path.isdir(path):
+            return cls._from_directory(path, skill_selector)
+        else:
+            logger.debug("No imports found. Importing everything.")
+            return cls.empty()
+
+    @classmethod
+    def _from_file(cls, path: Text, skill_selector: "SkillSelector") -> "SkillSelector":
+        path = os.path.abspath(path)
+        if data.is_config_file(path) and os.path.exists(path):
```

Comment:
Change order of checks: `if os.path.exists(path) and data.is_config_file(path)`.